install/install.ps1

```diff
@@ -1,94 +1,203 @@
+
+    Write-Verbose "Linking ${latestDirectory} to ${extractedDirectory}"
+    cmd.exe /Q /C "mklink /J `"$latestDirectory`" `"$extractedDirectory`"" > $null
+    if (!$?) {
+        Write-Error "Linking failed!"
+    }
+
+    Write-Verbose "Removing ${zipFile}"
+    Remove-Item -Force $zipFile
+
+    # Permanently add the latest directory to the current user's $PATH (if it's not already there).
+    # Note that this will only take effect when a new terminal is started.
+    $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+    $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+    if ($currentPath -NotMatch 'AzureAuth') {
+        Write-Verbose "Updating `$env:PATH to include ${latestDirectory}"
+        $newPath = if ($null -eq $currentPath) {
+            "${latestDirectory}"
+        } else {
+            "${currentPath};${latestDirectory}"
+        }
+        setx PATH $newPath > $null
+    }
+
+    Write-Output "Installed azureauth $version!"
 
-Write-Verbose "Linking ${latestDirectory} to ${extractedDirectory}"
-cmd.exe /Q /C "mklink /J `"$latestDirectory`" `"$extractedDirectory`"" > $null
-if (!$?) {
-    Write-Error "Linking failed!"
 }
 
-Write-Verbose "Removing ${zipFile}"
-Remove-Item -Force $zipFile
-
-# Permanently add the latest directory to the current user's $PATH (if it's not already there).
-# Note that this will only take effect when a new terminal is started.
-$registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
-$currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
-if ($currentPath -NotMatch 'AzureAuth') {
-    Write-Verbose "Updating `$PATH to include ${latestDirectory}"
-    $newPath = if ($currentPath -Match $null) {
-        "${latestDirectory}"
+function Install-Post-0-4-0 {
+
+    Write-Verbose "Installing using post-0.4.0 method"
+
+    $repo = if ([string]::IsNullOrEmpty($Env:AZUREAUTH_REPO)) { 'AzureAD/microsoft-authentication-cli' } else { $Env:AZUREAUTH_REPO }
+    $releaseName = "azureauth-${version}-win10-x64"
+    $releaseFile = "${releaseName}.zip"
+    $releaseUrl = "https://github.com/${repo}/releases/download/${version}/$releaseFile"
+
+    $azureauthDirectory = if ([string]::IsNullOrEmpty($Env:AZUREAUTH_INSTALL_DIRECTORY)) {
+        ([System.IO.Path]::Combine($Env:LOCALAPPDATA, "Programs", "AzureAuth"))
     } else {
-        "${currentPath};${latestDirectory}"
+        $Env:AZUREAUTH_INSTALL_DIRECTORY
     }
-    Set-ItemProperty -Path $registryPath -Name PATH -Value $newPath
+
+    $targetDirectory = ([System.IO.Path]::Combine($azureauthDirectory, $version))
+    $zipFile = ([System.IO.Path]::Combine($azureauthDirectory, $releaseFile))
+
+    Write-Verbose "Creating ${azureauthDirectory}"
+    $null = New-Item -ItemType Directory -Force -Path $azureauthDirectory

```

Comment:
This is being used to suppress output. It is functionally the same as redirecting output to `$null` with
```powershell
thing-that-does-output > $null
```
We used this originally because this was shown in examples that suppressed the output of built-in PowerShell commandlets, but I'm not sure if we could replace it with `> $null`. Never tried. 😅

Reply:
Why do we need to suppress it? Just curious!